test/puppetlabs/puppetdb/admin_clean_test.clj

```diff
@@ -0,0 +1,156 @@
+(ns puppetlabs.puppetdb.admin-clean-test
+  (:require [clj-http.client :as http-client]
+            [clojure.math.combinatorics :refer [combinations]]
+            [clojure.test :refer :all]
+            [metrics.counters :as counters]
+            [metrics.gauges :as gauges]
+            [metrics.timers :as timers]
+            [overtone.at-at :as at-at]
+            [puppetlabs.puppetdb.cheshire :as json]
+            [puppetlabs.puppetdb.cli.services :as cli-svc]
+            [puppetlabs.puppetdb.http :as http]
+            [puppetlabs.puppetdb.testutils.services :as svc-utils
+             :refer [*server* with-single-quiet-pdb-instance]]
+            [puppetlabs.puppetdb.utils :as utils]
+            [puppetlabs.trapperkeeper.app :refer [get-service]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]])
+  (:import
+   [java.util.concurrent CyclicBarrier]))
+
+(defn- post-admin [path form]
+  (http-client/post (str (utils/base-url->str (svc-utils/pdb-admin-url))
+                         "/"
+                         path)
+                    {:body (json/generate-string form)
+                     :throw-exceptions false
+                     :content-type :json
+                     :character-encoding "UTF-8"
+                     :accept :json}))
+
+(defn- checked-admin-post [path form]
+  (let [result (post-admin path form)]
+    (is (= http/status-ok (:status result)))
+    (when-not (= http/status-ok (:status result))
```

Comment:
If you mean the printing, then yes, I meant to keep it; otherwise, when something goes wrong, the tests just silently deadlock.